symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -137,47 +138,61 @@ public ApiClientWebClient(final WebClient webClient, String basePath, Map<String
       requestBodySpec.body(BodyInserters.fromValue(body));
     }
 
-    Mono<ClientResponse> mono = requestBodySpec.exchange();
-    ClientResponse response = mono.block();
+    try {
+      return requestBodySpec.exchangeToMono(response -> toApiResponse(returnType, response))
+          .block();
+    } catch (Exception e) {
+      Throwable unwrap = Exceptions.unwrap(e);
+      if (unwrap instanceof ApiException) {
+        throw (ApiException) unwrap;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private <T> Mono<ApiResponse<T>> toApiResponse(TypeReference<T> returnType, ClientResponse response) {
     Map<String, List<String>> headers = response
         .headers().asHttpHeaders().entrySet()
         .stream()
         .collect(Collectors.toMap(
             Map.Entry::getKey,
             Map.Entry::getValue
         ));
-    for (Map.Entry<String, List<String>> entry : response.headers().asHttpHeaders().entrySet()) {
-      headers.put(entry.getKey(), entry.getValue());
-    }
 
     if (response.statusCode().equals(HttpStatus.NO_CONTENT)) {
-      return new ApiResponse<>(response.statusCode().value(), headers);
+      return Mono.just(
+          new ApiResponse<>(response.statusCode().value(), headers));
     } else if (response.statusCode().equals(HttpStatus.OK)) {
```

Comment:
I was wondering what if the response is 201? that means we handle it as an error?

Reply:
The first `if` is `response.statusCode().equals(HttpStatus.NO_CONTENT)`. Sounds good right?

Reply:
NO_CONTENT is 204 right 
And there are a lot of others HTTP 2xx statuses too

Reply:
Oh my bad !

Reply:
Could you change the logic to `response.statusCode().is2xxSuccessful()`?

Reply:
Done